packages/client/src/connection.ts

```diff
@@ -100,6 +100,14 @@ export interface ConnectionOptions {
    */
   metadata?: Metadata;
 
+  /**
+   * API key for Temporal. This becomes the "Authorization" HTTP header with "Bearer " prepended.
+   * This is only set if RPC metadata doesn't already have an "authorization" key.
```

Comment:
> This is only set if RPC metadata doesn't already have an "authorization" key.

This is no longer true. `apiKey` and `authorization` header are mutually exclusive. Also add a mention for this on `metadata`'s doc.